src/app/clusters/operational-credentials/operational-credentials.cpp

```diff
@@ -33,33 +37,218 @@
+*/
+
+EmberAfStatus writeFabricAttribute(uint8_t * buffer, int32_t index = -1)
+{
+    EmberAfAttributeSearchRecord record;
+    record.endpoint         = 0;
+    record.clusterId        = ZCL_OPERATIONAL_CREDENTIALS_CLUSTER_ID;
+    record.clusterMask      = CLUSTER_MASK_SERVER;
+    record.manufacturerCode = EMBER_AF_NULL_MANUFACTURER_CODE;
+    record.attributeId      = ZCL_FABRICS_ATTRIBUTE_ID;
+
+    // When reading or writing a List attribute the 'index' value could have 3 types of values:
+    //  -1: Read/Write the whole list content, including the number of elements in the list
+    //   0: Read/Write the number of elements in the list, represented as a uint16_t
+    //   n: Read/Write the nth element of the list
+    //
+    // Since the first 2 bytes of the attribute are used to store the number of elements, elements indexing starts
+    // at 1. In order to hide this to the rest of the code of this file, the element index is incremented by 1 here.
+    // This also allows calling writeAttribute() with no index arg to mean "write the length".
+    
+    return emAfReadOrWriteAttribute(&record, NULL, buffer, 0, true, index + 1);
+}
+
+EmberAfStatus writeFabric(FabricId fabricId, NodeId nodeId, uint16_t vendorId, int32_t index)
+{
+    EmberAfStatus status    = EMBER_ZCL_STATUS_SUCCESS;
+
+    EmberAfFabricDescriptor fabricDescriptor;
+    fabricDescriptor.FabricId = fabricId;
+    fabricDescriptor.NodeId   = nodeId;
+    fabricDescriptor.VendorId = vendorId;
+
+    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Writing admin into attribute store at index %d: fabricId %" PRIX64 
+                            ", nodeId %" PRIX64 " vendorId %" PRIX16, index, fabricId, nodeId, vendorId);
+    status =  writeFabricAttribute((uint8_t *) &fabricDescriptor, index);
+    return status;
+}
+
+CHIP_ERROR writeAdminsIntoFabricsListAttribute(void)
+{
+    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Call to writeAdminsIntoFabricsListAttribute");
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    // Loop through admins
+    int32_t fabricIndex = 0;
+    for (auto & pairing : GetGlobalAdminPairingTable())
+    { 
+        NodeId nodeId = pairing.GetNodeId();
+        uint64_t fabricId = pairing.GetFabricId();
+        uint16_t vendorId = pairing.GetVendorId();
+
+        // Skip over uninitialized admins
+        if (nodeId != kUndefinedNodeId && fabricId != kUndefinedFabricId && vendorId != kUndefinedVendorId)
+        {
+            if (writeFabric(fabricId, nodeId, vendorId, fabricIndex) != EMBER_ZCL_STATUS_SUCCESS)
+            {
+                emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Failed to write admin with fabricId %" PRIX64 " in fabrics list", fabricId);
+                err = CHIP_ERROR_PERSISTED_STORAGE_FAILED;
+                break;
+            }
+        }
```

Comment:
```suggestion
            }
            fabricIndex ++;
        }
```
because I don't see where we are incrementing it now....

Reply:
fixed, needed to do another round of testing as I had mentioned :)